django_project/cplus_api/models/layer.py

```diff
@@ -131,6 +132,16 @@ def download_to_working_directory(self, base_dir):
                 destination.write(chunk)
         self.last_used_on = timezone.now()
         self.save(update_fields=['last_used_on'])
+        if file_path.endswith('.zip'):
+            with ZipFile(file_path, 'r') as zip_ref:
+                zip_ref.extractall(dir_path)
+            shapefile = [
+                file for file in os.listdir(dir_path) if file.endswith('.shp')
```

Comment:
@zamuzakki can we find the correct shapefile by using the zip filename?
e.g. the zip is test_layer.zip, so the correct shp filename would be test_layer.shp
This is to ensure that this code returns correct shapefile when there are more than 1 extracted zip file.

Reply:
@danangmassandy That was what I did at first. But I was thinking about cases where the zip file name is different from the shapefile name. I'll update the extract location to a sub-directory, so each zip file would be extracted to different directory inside `dir_path`.